src/OmniSharp.Roslyn/OmniSharpWorkspace.cs

```diff
@@ -83,6 +86,19 @@ public void AddDocument(DocumentInfo documentInfo)
             OnDocumentAdded(documentInfo);
         }
 
+        public DocumentId AddMiscellanousFileDocument(ProjectId projectId, string filePath)
```

Comment:
Given that this workspace already understands how to handle misc files, is there a reason clients need to be able to control what project they wind up in?

Reply:
In that case, we will have to add a "MiscFileProject" to the workspace and the first time we see a misc file (or even in the constructor of the MiscFileSystem) we will have to pass it to the workspace. From then on, every misc file will be added to the same project. Does that seem a better approach ?